### ISSUE TITLE
cf push -p fails when target is a symlink

### ISSUE THREAD

**Issue by geofffranks** [cloudfoundry/cli#634]
If I run `cf push myApp -p /path/to/my/app`, and `/path/to/my/app` is a symlink pointing to another directory, `cf push` fails, with a message similar to the following:

```
  "entity": {
    "guid": "998342f6-2b72-407a-8e70-f0a68d3d38bc",
    "status": "failed",
    "error": "Use of entity>error is deprecated in favor of entity>error_details.",
    "error_details": {
      "code": 150001,
      "description": "The app package is invalid: Could not zip the package\n STDOUT: \"\nzip error: Nothing to do! (try: zip -q -r --symlinks /var/vcap/data/cloud_controller_ng/tmp/safezipper20151015-2739-n84n09/package.zip . -i .)\n\"\n STDERR: \"\"",
      "error_code": "CF-AppPackageInvalid"
    }
  }
```

What I believe is happening is that the symlink is not being followed to its target, and so the symlink itself is being zipped and uploaded to the cloud controller, which doesn't actually contain the app source. 

I'm working around by not using symlinks, but figured I should submit this to get the correct behavior in `cf push`

**Comment by nota-ja** [comment]
As far as I know, symlinks are ignored intentionally.

- https://github.com/cloudfoundry/cli/issues/108
- https://github.com/cloudfoundry/cli/blob/3ce3f3857de21435d986f51b74f474ac8267520a/cf/app_files/app_files.go#L95-L97

**Comment by geofffranks** [comment]
That looks like it's regarding files inside the app dir that are symlinks. This issue was about the appdir itself being a symlink. It seems like in https://github.com/cloudfoundry/cli/blob/3ce3f3857de21435d986f51b74f474ac8267520a/cf/commands/application/push.go#L607-L655 the app dir is just assumed to be a directory, but if it's a symlink, the link itself is zipped and uploaded, rather than the target of the link.  Or perhaps if it's a symlink, it is assumed to be a zip file rather than a directory?

**Comment by dkoper** [comment]
Hi @geofffranks 

I believe we have addressed this in our recent cf CLI 6.15.0 release.
https://github.com/cloudfoundry/cli#downloads

Thanks,
Dies Koper
CF CLI PM
